Arrowgene.Ddon.GameServer/BazaarManager.cs

```diff
@@ -158,7 +154,16 @@ public uint ReceiveProceeds(GameClient client)
             foreach (BazaarExhibition exhibition in exhibitionsToReceive)
             {
                 exhibition.Info.State = BazaarExhibitionState.Idle;
-                exhibition.Info.Expire = now.Add(COOLDOWN_TIME_SPAN);
+                ulong totalCooldown;
+                try
+                {
+                    totalCooldown = Server.Setting.GameLogicSetting.BazaarCooldownTimeSeconds - Server.GpCourseManager.BazaarReExhibitShorten();
```

Comment:
I feel you could've just checked to see if the shorten time is larger than the time left but I guess this works.